handlers/handlers.go

```diff
@@ -14,126 +13,92 @@ import (
 const FilePath = "assets/pokemon.csv"
 
 func GetAll(w http.ResponseWriter, r *http.Request) {
-	pokemonMap := make(map[int]string)
-	w.Header().Set("Content-Type", "application/json")
-
-	file, err := os.Open(FilePath)
+	records, err := usecase.ReadCsv(FilePath)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(err.Error()))
+		usecase.HandleError(w, err, http.StatusInternalServerError)
 		return
 	}
-	reader := csv.NewReader(file)
-	records, err := reader.ReadAll()
+	response, err := usecase.ParseRecordsToJSON(records)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("CSV is invalid"))
+		usecase.HandleError(w, err, http.StatusInternalServerError)
 		return
 	}
-	for _, pokemon := range records {
-		id, err := strconv.Atoi(pokemon[0])
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte("Error parsing ID"))
-			return
-		}
-		pokemonMap[id] = pokemon[1]
-	}
-	response, err := json.Marshal(pokemonMap)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("Error parsing result"))
-		return
-	}
-	w.WriteHeader(http.StatusOK)
-	w.Write(response)
+	usecase.HandleSuccess(w, response)
 }
 
 func GetById(w http.ResponseWriter, r *http.Request) {
 	pathParams := mux.Vars(r)
-	pokemonMap := make(map[string]string)
-	pokemonId := pathParams["pokemonId"]
-	w.Header().Set("Content-Type", "application/json")
-
-	file, err := os.Open(FilePath)
+	pokemonId, _ := strconv.Atoi(pathParams["pokemonId"])
+	records, err := usecase.ReadCsv(FilePath)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("Error reading CSV file"))
+		usecase.HandleError(w, err, http.StatusInternalServerError)
 		return
 	}
-	reader := csv.NewReader(file)
-	records, err := reader.ReadAll()
+	pokemonMap, err := usecase.ParseRecordsToMap(records)
+
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("CSV is invalid"))
+		usecase.HandleError(w, err, http.StatusInternalServerError)
 		return
 	}
-	for _, pokemon := range records {
-		pokemonMap[pokemon[0]] = pokemon[1]
-	}
-
 	if _, ok := pokemonMap[pokemonId]; ok {
 		result := make(map[string]string)
-		result["id"] = pokemonId
+		result["id"] = strconv.Itoa(pokemonId)
 		result["name"] = pokemonMap[pokemonId]
 		response, err := json.Marshal(result)
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte("Error parsing result"))
+			usecase.HandleError(w, err, http.StatusInternalServerError)
 			return
 		}
-		w.WriteHeader(http.StatusOK)
-		w.Write(response)
+		usecase.HandleSuccess(w, response)
+		return
 	} else {
-		w.WriteHeader(http.StatusNotFound)
-		w.Write([]byte("Resource Not Found with id"))
+		err = errors.New("Resource Not Found")
+		usecase.HandleError(w, err, http.StatusNotFound)
 	}
```

Comment:
Since you have a `return` at the end of your previous `if`, you can drop this `else`.